overlord/hookstate/hooks_test.go

```diff
@@ -316,7 +337,7 @@ func (s *gateAutoRefreshHookSuite) TestGateAutorefreshHookError(c *C) {
 	c.Assert(change.Status(), Equals, state.DoneStatus)
 
 	// and snap-b is now held.
-	checkIsHeld(c, st, "snap-b", "snap-a")
+	checkIsHeld(c, st, "snap-a", "snap-a")
```

Comment:
this changes are strange? why are they happening? also the comment here needs updating too

Reply:
I'm still confused by this bit? can you elaborate?

Reply:
This is because previously - when all the affecting/affected info was statically set on the hook context upfront - the test could easily create fake dependencies with e.g. `task := hookstate.SetupGateAutoRefreshHook(st, "snap-a", false, false, map[string]bool{"snap-b": true})` (see above, now deleted). 

With changes in this PR these tests now require proper mocking of refresh-candidates and real snap-interdependencies (via plug/slot or base snap), so for this test I opted for a single snap that just holds itself. I think what may add to the confusion is that previously this test should actually also do `checkIsHeld(c, st, "snap-a", "snap-a")`, aside from `checkIsHeld(c, st, "snap-b", "snap-a")`.